### DISCUSSION TITLE
Licensing choice.

### DISCUSSION THREAD

**Discussion by delaneyj** [rhashimoto/wa-sqlite#41] [Q&A] [answered]
Building your library and using it seems to working wonderfully.  I'd love to use in projects going forward but I don't know how that is possible with the GPLv3.  IANAL, but given the viral nature of GPL it would mean the entire app has to be licensed GPL; given the library is part of the rollup/vite bundle.  If its a matter of creating PRs for any improvements/changes to the library I'd be happy to make that conditional.  If there is a deeper concern I'd really like to know as the current license will hamper its usage.

Specifically it would be great to see it available under the [APL](https://choosealicense.com/licenses/apache-2.0/) as it limits the changes to those directly effecting wa-sqlite.

**Comment by rhashimoto** [comment, marked as answer]
Thanks for your interest! The GPL was chosen as the default license specifically for its viral nature. In my view, that best allows anyone to experiment with the package, while preserving some commercial licensing opportunities. This is also the reason why [unsolicited pull requests are discouraged](https://github.com/rhashimoto/wa-sqlite/blob/master/.github/pull_request_template.md) (forks are not, though they are still bound by the GPL).

I understand that the terms of the GPL may inhibit wider adoption and deployment. Widespread community usage is not a project priority (as also evidenced by the [lack of npm registration](https://github.com/rhashimoto/wa-sqlite/issues/12) 😀). There are a few more disadvantages with this package to consider [here](https://github.com/lana-k/sqliteviz/issues/45#issuecomment-874278192).

Anyone interested in commercial licensing can contact licensing@shoestringresearch.com.

**Comment by delaneyj** [comment]
That's unfortunate.  The issue is GPLv3 is trying to override all other libraries licenses.  I understand your stance and hope you can find an audience.  If you are trying to make commercial licensing an avenue it'd be nice to have a site or discussion on pricing.

**Reply by rhashimoto** [reply]
FYI the project has [switched to the MIT License](https://github.com/rhashimoto/wa-sqlite/discussions/75) if you're still interested.
